com.reprezen.swagedit/src/com/reprezen/swagedit/utils/SwaggerFileFinder.java

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright (c) 2016 ModelSolv, Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ModelSolv, Inc. - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package com.reprezen.swagedit.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceProxy;
+import org.eclipse.core.resources.IResourceProxyVisitor;
+import org.eclipse.core.runtime.CoreException;
+
+import com.google.common.collect.Lists;
+
+public class SwaggerFileFinder {
+
+    /**
+     * Represents the scope for which the JSON reference proposals have to be computed. <br/>
```

Comment:
`Scope` was moved from `JsonReferenceProposalProvider` and now serves several purposes,  it's not limited to reference proposals anymore. The javadoc should be updated accordingly.